torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -782,6 +783,11 @@ std::shared_ptr<ProcessGroupNCCL::WorkNCCL> ProcessGroupNCCL::initWork(
   return std::make_shared<ProcessGroupNCCL::WorkNCCL>(devices);
 }
 
+std::vector<at::Tensor> ProcessGroupNCCL::WorkNCCL::result() {
+  TORCH_CHECK(isCompleted());
```

Comment:
This is not necessarily true for NCCL. `isCompleted()` returns true only when the work has completed successfully on the GPU. But we can still retrieve the `result` for the Work and perform async processing. I think we shouldn't have this check for NCCL.

Reply:
Good point, removed.